service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerAboutToSubmitTest.java

```diff
@@ -128,6 +132,47 @@ void aboutToSubmitShouldNotHaveDraftAppendedToFilename() {
         assertThat(caseData.getOrderCollection().get(0).getValue().getDocument()).isEqualTo(expectedDocument());
     }
 
+    @Test
+    void shouldAddCloseCaseObjectToCaseDataWhenOptionWasSelectedAndChangeState() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = buildJudgeAndLegalAdvisor(NO);
+
+        final CaseDetails caseDetails = buildCaseDetails(
+            commonCaseDetailsComponents(CARE_ORDER, FINAL, judgeAndLegalAdvisor)
+                .caseLocalAuthority(LOCAL_AUTHORITY_CODE)
+                .children1(createChildren("Fred", "John"))
+                .orderAppliesToAllChildren("Yes")
+                .closeCaseFromOrder("Yes"));
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(caseDetails);
+        CloseCase closeCase = mapper.convertValue(response.getData().get("closeCaseTabField"), CloseCase.class);
+        State state = mapper.convertValue(response.getData().get("state"), State.class);
+
+        CloseCase expected = CloseCase.builder()
+            .date(dateNow())
+            .showFullReason(YES)
+            .reason(FINAL_ORDER)
+            .build();
+
+        assertThat(closeCase).isEqualTo(expected);
+        assertThat(state).isEqualTo(CLOSED);
+    }
+
+    @Test
+    void shouldAddCloseCaseObjectToCaseDataWhenOptionWasNotSelected() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = buildJudgeAndLegalAdvisor(NO);
+
+        final CaseDetails caseDetails = buildCaseDetails(
+            commonCaseDetailsComponents(CARE_ORDER, FINAL, judgeAndLegalAdvisor)
+                .caseLocalAuthority(LOCAL_AUTHORITY_CODE)
+                .children1(createChildren("Fred", "John"))
```

Comment:
see my PR #1260 for something I added to TestDataHelper: testChildren() - I'd prefer using that over having a separate private method in each test class which does the same thing (since you're already editing these controller tests maybe can swap createChildren() out for testChildren())

Reply:
Will have a look